torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -946,6 +969,110 @@ void LoopNest::splitWithMask(For* f, int factor, For** outer, For** inner) {
   // TODO: record history of transformations
 }
 
+void LoopNest::reorderAxis(Tensor* t, For* a, For* b) {
```

Comment:
We don't really use the tensor here, only its loops: maybe take `std::vector<For*>` as the first parameter instead (on the callsite we'll get it by `l.getLoopStmtsFor(t)`)?

Reply:
Not sure about this. It would save the extra call to getLoopStatementsFor but it's not longer really a loopNest thing. Prefer to refactor when we pull this and maybe rfactor/split out

Reply:
I was thinking about it mostly from the data we're accessing here: we pass the tensor, but only use part of it that's responsible for the corresponding loops. That made me think that we should pass only the loops since we're not using anything else from the tensor. I don't have a strong preference on it though since we don't have clear guidelines on it yet anyways :)